CMakeLists.txt

```diff
@@ -1,32 +1,80 @@
-cmake_minimum_required(VERSION 3.1)
-
-
-set(CMAKE_DISABLE_SOURCE_CHANGES ON)
-set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+cmake_minimum_required(VERSION 3.9)
```

Comment:
I already have trouble with 3.1 being too new on some platforms :) 
Is 3.9 strictly needed or can we do with older versions?

Reply:
Even worse, it looks like FetchContent is only available since 3.11 :) ... We can use ExternalProject instead, but then the download etc. will happen at build time, not configure time. If that's not a problem, I can try to lower the required version. On the other hand, if 3.1 is already too new, does it really matter which version we use? I.e. one has to compile CMake from scratch anyway, they most likely will use the latest (3.12) version.